generic/curl-insecure.yaml

```diff
@@ -0,0 +1,22 @@
+rules:
+  - id: curl-insecure
+    message: Found `curl` command disabling SSL verification
+    languages: [generic]
+    severity: WARNING
+    metadata:
+      category: security
+      subcategory: [audit]
+      technology: [shell]
+      cwe: "CWE-295: Improper Certificate Validation"
+      confidence: MEDIUM
+      likelihood: MEDIUM
+      impact: HIGH
+      references:
+        - https://curl.se/docs/manpage.html
+    pattern-either:
+      # A space character was left at the end of some patterns to help ensure
+      # that the intended flag was used, and minimize the chance that another,
+      # longer flag that _starts with_ the intended flag results in a false
+      # positive
+      - pattern: "curl ... -k "
+      - pattern: "curl ... --insecure"
```

Comment:
Random thought/comment: one pattern ends with space, the other one doesn't; is that ok?

Reply:
That was my feeble attempt to remove some potential false positives 😆 

Without the extra space, this would catch things like `curl -ksomeotherflag ...` or `curl somethingsomething | othercommend -ksomeotherflag`. I think these situations are very unlikely, but generic mode can produce a lot of false positives, so I was trying to minimize that as much as possible.